source/lac/petsc_parallel_block_sparse_matrix.cc

```diff
@@ -33,8 +33,9 @@ namespace PETScWrappers
 
 
     void
-    BlockSparseMatrix::reinit(const size_type n_block_rows,
-                              const size_type n_block_columns)
+    BlockSparseMatrix::reinit(
+      const BlockSparseMatrix::size_type n_block_rows,
+      const BlockSparseMatrix::size_type n_block_columns)
```

Comment:
Just undo this change. Once you're past the opening `(`, you're in scope for the class, and you don't need to qualify `size_type` with the class name any more.

Reply:
I will remove this change, as doxygen still complains. But it looks like it gives us more insight what's going wrong.
```
/home/runner/work/dealii/dealii/source/lac/petsc_parallel_block_sparse_matrix.cc:36: warning: no uniquely matching class member found for 
  void BlockSparseMatrix< number >::reinit(unsigned int n_block_rows, const BlockSparseMatrix::size_type n_block_columns)
```
`n_block_rows` is an `unsigned int`, but `n_block_columns` is a `size_type`?

Reply:
I have a strong feeling that `n_block_rows` gets confused with the member function [`n_block_rows`](https://www.dealii.org/developer/doxygen/deal.II/classBlockMatrixBase.html#a2478164707bf2932ad32ddd27babfc00).

There is also a function `n_block_cols`, but the argument in the `reinit` function is `n_block_columns` so it is not ambiguous.

Reply:
The `Trilinos` equivalent has simply been hidden from `doxygen` in #9880. I will do the same here.
https://github.com/dealii/dealii/blob/6c065836a1ff9259393f6e851c60b20ec0cddb13/source/lac/trilinos_block_sparse_matrix.cc#L41-L67